### ISSUE TITLE
Performances Issue

### ISSUE THREAD

**Issue by ssram50** [leaderj1001/LambdaNetworks#1]
I have run the code as is, and for the default LambdaReset18 model
I am seeing accuracy of only 71.39% on the test dataset (which 96% on the training dataset)
Could you share the model for LambdaReset18

**Comment by leaderj1001** [comment]
Thanks for comments !

You need to replace a code.
https://github.com/leaderj1001/LambdaNetworks/blob/95278e28fda14b9034a30be5bfecdfd57fef5095/model.py#L124

```python
self.conv1 = nn.Conv2d(3, 64, kernel_size=3, stride=1, padding=1, bias=False)
```

**Comment by ssram50** [comment]
Accuracy has improved a lot after reducing the kernel size from 7 to 3
But the max accuracy is still only 89.10% which is quite far from what is reported 93.32%

The model Parameters: 10121128 is different than what is reported 8.6M.
Is there any additional architecture changes which are required to add to the code

Also the default number of class is1000:
def __init__(self, block, num_blocks, num_classes=1000):
Which slows down the convergence, better to set it to 10 for CIFAR10

Thanks for the support!

Current Learning Rate: [1.9637012099169403e-05]
[Epoch:  310], Loss: 0.000, Acc: 100.000, Correct 12928.0 / Total 12928.0
[Epoch:  310], Loss: 0.000, Acc: 100.000, Correct 25728.0 / Total 25728.0
[Epoch:  310], Loss: 0.000, Acc: 100.000, Correct 38528.0 / Total 38528.0
**[Epoch:  310], Acc: 89.110**

**Comment by leaderj1001** [comment]
The uploaded code is for ImageNet training. Therefore, you need to convert the code according to CIFAR to get the desired performance. In addition, 8.6M is a parameter when classes=10.

Thanks.

**Comment by ssram50** [comment]
When setting the number of classes=10
The number of reported parameters is:
Model Parameters: 8092618

**Comment by leaderj1001** [comment]
I think your u, m parameter settings are different.
u=1, m=23, 8.0M
u=4, m=7, 8.6M

**Comment by ssram50** [comment]
I fixed values for u and m
and the model size now matches.

The performances for: LambdaResNet18 are still lower than expected:
Could you think of a reason why that may happen?


Current Learning Rate: [0.0654543046337755]
[Epoch:  215], Loss: 0.070, Acc: 97.625, Correct 12621.0 / Total 12928.0
[Epoch:  215], Loss: 0.075, Acc: 97.392, Correct 25057.0 / Total 25728.0
[Epoch:  215], Loss: 0.073, Acc: 97.485, Correct 37559.0 / Total 38528.0
[Epoch:  215], Acc: 86.770

Thanks

**Comment by ssram50** [comment]
Resnet18 performance are verified (very close)

Epoch:  147], Loss: 0.002, Acc: 99.992, Correct 12927.0 / Total 12928.0
[Epoch:  147], Loss: 0.002, Acc: 99.992, Correct 25726.0 / Total 25728.0
[Epoch:  147], Loss: 0.002, Acc: 99.992, Correct 38525.0 / Total 38528.0
[Epoch:  147], Acc: 92.800 (compared to 93.0)

**Comment by ssram50** [comment]
Performances for:
LambdaResNet18(num_classes=10)
Have not being verified yet:

Current Learning Rate: [1.9637012099169403e-05]
[Epoch:  310], Loss: 0.001, Acc: 99.996, Correct 25855.0 / Total 25856.0
[Epoch:  310], Acc: 86.650

**Comment by naviocean** [comment]
@leaderj1001 I got the same Acc with @ssram50, its still lower than your AP:
```
020-11-24 13:37:47,301 Lambda INFO: [Epoch:   77], Loss: 0.015, Acc: 99.520, Correct 12866.0 / Total 12928.0
2020-11-24 13:37:58,849 Lambda INFO: [Epoch:   77], Loss: 0.015, Acc: 99.522, Correct 25605.0 / Total 25728.0
2020-11-24 13:38:10,404 Lambda INFO: [Epoch:   77], Loss: 0.015, Acc: 99.541, Correct 38351.0 / Total 38528.0
2020-11-24 13:38:23,640 Lambda INFO: [Epoch:   77], Acc: 87.450
```

**Comment by leaderj1001** [comment]
When I use u=4, m=7, I get a 94.2% Acc.
  - without Dropout,
  - Gradient Clip: 2.0
  - stem: self.conv1 = nn.Conv2d(3, 64, kernel_size=3, stride=1, padding=1, bias=False)
![lambdanet_result](https://user-images.githubusercontent.com/22078438/100063384-81945c80-2e74-11eb-8ad1-71606d6013c6.jpeg)
Since the CIFAR-10 32x32 image resolution is used, it seems that u=4 and m=7 show better results.

Thank you.

**Comment by naviocean** [comment]
@leaderj1001 u=1, m=23 are settings for exp above.
Could you please share your code for CIFAR-10 or create new branch CIFAR-10.
Thanks

**Comment by changlin31** [comment]
Hi, @ssram50 @naviocean 
I've successfully achieved **93.95 Acc** with LambdaReset18. In addition to the setting of @leaderj1001 :

> * u=4, m=7,
> * without Dropout,
> * Gradient Clip: 2.0
> * stem: self.conv1 = nn.Conv2d(3, 64, kernel_size=3, stride=1, padding=1, bias=False)

I remove the first pooling layer by commenting out these lines: https://github.com/leaderj1001/LambdaNetworks/blob/3a51b6a7dbac9490dffed6a1bf68157e93135da6/model.py#L129
https://github.com/leaderj1001/LambdaNetworks/blob/3a51b6a7dbac9490dffed6a1bf68157e93135da6/model.py#L159
After this I achieved **93.27 Acc**.
```
[Epoch:  146], Loss: 0.000, Acc: 100.000, Correct 12928.0 / Total 12928.0
[Epoch:  146], Loss: 0.000, Acc: 100.000, Correct 25728.0 / Total 25728.0
[Epoch:  146], Loss: 0.000, Acc: 99.997, Correct 38527.0 / Total 38528.0
[Epoch:  146], Acc: 93.270
Best Model Saving...
Current Learning Rate: 0.000714967387272874
```

By replacing CosineAnnealingWarmRestarts in:
https://github.com/leaderj1001/LambdaNetworks/blob/3a51b6a7dbac9490dffed6a1bf68157e93135da6/main.py#L97
with: `lr_scheduler = torch.optim.lr_scheduler.CosineAnnealingLR(optimizer, T_max=200)`
I further achieved **93.95 Acc**.
```
[Epoch:  209], Loss: 0.000, Acc: 100.000, Correct 12928.0 / Total 12928.0
[Epoch:  209], Loss: 0.000, Acc: 100.000, Correct 25728.0 / Total 25728.0
[Epoch:  209], Loss: 0.000, Acc: 100.000, Correct 38528.0 / Total 38528.0
[Epoch:  209], Acc: 93.950
Best Model Saving...
Current Learning Rate: 0.0004988171141721232
```

**Comment by ssram50** [comment]
Hi @changlin31
Can you please share your code for the version that achieves: 93.95 Acc.
Thanks

**Comment by changlin31** [comment]
Hi @ssram50 ,
I further achieved 94.12 Acc. Check out the modified code here: https://github.com/changlin31/LambdaNetworks/tree/cifar10
cheers

**Comment by ssram50** [comment]
Hi @changlin31
Thank you so much for you help :) 
I have verified the performances for CIFAR10 with Acc 93.470
![LambdaNetworkCIFAR10](https://user-images.githubusercontent.com/2352259/102372758-6220b900-3f74-11eb-899b-dd7d3f55c037.png)

**Comment by ssram50** [comment]
Attaching another snapshot for the performances
Acc = 94.02

![res2](https://user-images.githubusercontent.com/2352259/102635049-98db0880-4107-11eb-80ba-790408453bc4.png)
